core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/dbcreator/DatabaseCreator.java

```diff
@@ -349,6 +350,30 @@ private void executeSQLScript() throws Exception {
         }
     }
 
+    private String getDatabaseTypeWithVersion(String databaseType) throws SQLException {
+        String databaseProductVersion = null;
+        try {
+            if (databaseType != null && databaseType.equals("mysql")) {
+                if (conn != null && (!conn.isClosed())) {
+                    DatabaseMetaData metaData = conn.getMetaData();
+                    databaseProductVersion = metaData.getDatabaseProductVersion();
+                    double versionWithMinorNumber = Double.parseDouble(databaseProductVersion.substring(0, 3));

```

Comment:
We better try to split by the second "." and get the version, else this logic misinterprets versions such as 5.10